File: problems/Valid Palindrome/isPalindrome.cs
public class Solution {
    public bool IsPalindrome(string s) {
        var n = s.Length;
        var l = 0;
        var r = n - 1;

        while (l < r) {
            while (l < r && !char.IsLetter(s[l]) && !char.IsDigit(s[l])) {++l;}
            while (l < r && !char.IsLetter(s[r]) && !char.IsDigit(s[r])) {--r;}

            if (l > r || char.ToLower(s[l++]) != char.ToLower(s[r--])) {
                return false;
            }
        }

        return true;
    }
}

File: problems/Valid Palindrome/isPalindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        StringBuilder sb = new StringBuilder();

        for (int idx = 0; s.length() > idx; ++idx) {
            if (Character.isLetterOrDigit(s.charAt(idx))) {
                sb.append(Character.toLowerCase(s.charAt(idx)));
            }
        }

        int left = 0;
        int right = sb.length() - 1;

        while (left < right) {
            if (sb.charAt(left++) != sb.charAt(right--)) {
                return false;
            }
        }

        return true;
    }
}
